content/articles/2018-10-devtools-2.0.0.Rmarkdown

```diff
@@ -0,0 +1,256 @@
+package.
+
+When used non-interactively the install functions work like the previous default of
+always upgrading outdated packages.
+
+<p align="center">
+<img src="/images/devtools-2.0.0/upgrade-menu.png" alt="upgrade menu">
+</p>
+
+The menu respects the `menu.graphics` option, so set `options(menu.graphics =
+FALSE)` in your .Rprofile if you prefer text based menus, or `TRUE` if you
+prefer graphical widgets.
+
+### Improved check output
+
+`check()` now uses the [rcmdcheck] package, which has much richer (and
+colorful) output to the check results, which makes finding where a check fails
+much easier.
+
+<p align="center">
+<a href="https://github.com/r-lib/rcmdcheck">
+<img src="https://raw.githubusercontent.com/r-lib/rcmdcheck/master/inst/rcmdcheck.gif" alt="rcmdcheck output" width="700">
+</a>
+</p>
+
+### Testing single files
+
+devtools now includes functions (`test_file()` and `test_coverage_file()`) to
+improve development of a single file. Rather than running all
+tests, or manually supplying a `filter` argument to restrict the tests
+`test_file()` automatically runs the corresponding tests for a given source
+file. These functions make the feedback loop when developing new features
+quicker as you only run the relevant tests for the file you are editing.
+
+This requires you use a [standard naming
+convention](http://style.tidyverse.org/tests.html) for your tests, e.g. if you
+have a source file `R/featureA.R` the corresponding test file would be
+`tests/testthat/test-featureR.R`.
+
+The tests file to run is automatically detected from the open file in RStudio (if
+available), so you can call `test_file()` with either the source file or
+the test file open. A corresponding `test_coverage_file()`
+function shows the test code coverage for a single source file.
+
+There is also a `test_coverage()` function to report test coverage for your
+whole package.
+
+These functions have [RStudio addins](https://rstudio.github.io/rstudioaddins/)
+which allows you to [ bind them to shortcut
+keys](https://rstudio.github.io/rstudioaddins/#keyboard-shorcuts).
+
+Shortcuts we recommend
+
+<style>
+kbd {
+  border: 1px solid #aaa;
+  border-radius: 0.2em;
+  background-color: #f9f9f9;
+  padding: 0.1em 0.3em;
+}
+td,th {
+  padding: 0.4em;
+}
+</style>
+
+| Function                 | Windows shortcut                                | macOS shortcut                                        |
+| ------------------------ | ------------------------------------------------|------------------------------------------------------ |
+| `test_file()`            | <kbd>Ctrl</kbd>+<kbd>T</kbd>                    | <kbd>Command ⌘ </kbd>+<kbd>T</kbd>                    |
+| `test_coverage_file()`   | <kbd>Ctrl</kbd>+<kbd>G</kbd>                    | <kbd>Command ⌘ </kbd>+<kbd>G</kbd>                    |
+| `test_coverage()`        | <kbd>Ctrl</kbd>+<kbd>Shift ⇧</kbd>+<kbd>T</kbd> | <kbd>Command ⌘ </kbd>+<kbd>Shift ⇧</kbd>+<kbd>G</kbd> |
+
```

Comment:
Very slick!

Reply:
Cmd G ... is awfully standard for finding the next match, i.e. as a follow up to Cmd F

Reply:
The last row ... the recommended **letter** differs across platform. is that intentional?